subsys/net/ip/net_context.c

```diff
@@ -1378,6 +1594,36 @@ int net_context_get_option(struct net_context *context,
 	return ret;
 }
 
+struct net_context *net_context_find(sa_family_t family,
```

Comment:
Such a function doesn't exist already? How have we survived for so long without it?

Reply:
I think we have some confusion here. This function is added from the beginning in my PR. Only thing I split the patch. Tomasz was asking to split some patches because of various changes.

Reply:
> This function is added from the beginning in my PR. Only thing I split the patch.

Sounds good. But can you summarize in the commit message why we need such a function?

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/12564/commits/77880b96127455a6619e1bf172182c504bb4881d This helper need by this patch.

Reply:
> 77880b9 This helper need by this patch.

So, please use a commit message like:

~~~
Helper finds the matching network context based on family,
type and protocol: net_context_find(). This function will be
used in the implementation of AF_PACKET net_context, which
are by their nature "connectionless", so it's not convenient to use
existing struct net_conn machinery.
~~~

Btw, I still don't understand why you don't reuse net_conn, especially **after** you patched it to accept address family, but I don't go as far as thoroughly questioning that matter in the current situation (the PR being rushed for 1.14).

Reply:
Hmm, after all this API is not required at the moment. It was introduced for a reason. If user open a packet socket and udp/tcp socket. The same net_pkt has to be feed to both the sockets. So if user enabled CONFIG_NET_SOCKETS_PACKET and opened a packet socket, then packet has to be feed as it it. Then for udp socket data has to feed. May be we have to clone the packet. But If user enabled CONFIG_NET_SOCKETS_PACKET but doesn't opened any packet socket, then we need not to clone. So to find that particular context we need this helper.

After all in this initial version we don't need this at all. I will remove that patch.